pkgs/development/compilers/ghdl/default.nix

```diff
@@ -1,21 +1,25 @@
-{ stdenv, fetchurl, gnat, zlib }:
+{ stdenv, fetchurl, gnat, zlib, llvm_35, ncurses, clang, flavour ? "mcode" }:
 
-# I think that mcode can only generate x86 code,
-# so it fails to link pieces on x86_64.
-assert stdenv.system == "i686-linux";
+# mcode only works on x86, while the llvm flavour works on both x86 and x86_64.
+
+
+assert flavour == "llvm" || flavour == "mcode";
 
 let
+  optional = stdenv.lib.optional;
   version = "0.33";
 in
 stdenv.mkDerivation rec {
-  name = "ghdl-mcode-${version}";
+  name = "ghdl-${version}";
 
   src = fetchurl {
```

Comment:
While you're at it, maybe port this to `fetchFromGitHub`, it should be more stable when using github's generated tarballs.